### PR TITLE
Issue #110 - Added max attendee count as optional parameter for groups.

### LINKED ISSUE #110: Add maximum number of attendees to group
We can add another optional field to group, that can limit the number of people able to join to a group. If not set, it should allow infinite number of people.

### PR DESCRIPTION
**Pull request by vercibar**
Closes #110

### CODE REVIEW

File: src/components/groups/group.middlewares.ts [hunk 1]
```diff
@@ -131,10 +133,28 @@ export const validateGroup = () => {
...
+    check('maxAttendees', 'Legalább 1, maximum 100 fő vehet részt!')
+      .optional({ checkFalsy: true })
+      .isInt({ min: 1, max: 100 })
   ]
 }
 
+export const checkMaxAttendees = asyncWrapper(
+  async (req: Request, res: Response, next: NextFunction) => {
+    const users: User[] = await Group.relatedQuery('users').for(req.params.id)
+    if (users.length > (req.body.maxAttendees || 100)) {
```
Comment (triszt4n) [inline src/components/groups/group.middlewares.ts:146]: We do not have to query all the related users, we need the number of them, for which we could use:
```typescript
const userCount = await Group.relatedQuery('users').for(req.params.id).count()
```

This way, the query will be more goal-oriented and efficient.

File: src/components/groups/group.middlewares.ts [hunk 2]
```diff
@@ -13,9 +13,11 @@ export const joinGroup = asyncWrapper(async (req: Request, res: Response, next:
   const user = req.user as User
   const group = req.group
 
-  //Join group if not already in it, and it's not closed or it's the owner who joins.
+  // Join group if not already in it, and it's not closed or it's the owner who joins.
+  // We only join the group if it is not full already
   if ((!group.doNotDisturb || (user.id === group.ownerId)) &&
-    !group.users?.find(it => it.id === user.id)) {
+    !group.users?.find(it => it.id === user.id) &&
+    ((await Group.relatedQuery('users').for(req.group.id)).length < (group.maxAttendees || 100))) {
```
Comment (triszt4n) [inline src/components/groups/group.middlewares.ts:20]: See other comment:
>  We do not have to query all the related users, we need the number of them, for which we could use:
> ```typescript
> const userCount = await Group.relatedQuery('users').for(req.params.id).count()
> ```
> 
> This way, the query will be more goal-oriented and efficient.

File: migrations/20201012224257_add_max_attendees.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import * as Knex from "knex";
```
Comment (triszt4n) [inline migrations/20201012224257_add_max_attendees.ts:1]: My eslint has been angry with the code:
Screenshot 2020-10-13 01 05 10

**Review by triszt4n — CHANGES_REQUESTED** [review]
All in all, nice implementation. 👍 ✨

File: src/components/groups/group.routes.ts [hunk 4]
```diff
@@ -16,8 +16,8 @@ import {
   leaveGroup,
   createICSEvent,
   checkConflicts,
-  validateGroup, 
-  isGroupOwnerOrAdmin
+  validateGroup,
+  isGroupOwnerOrAdmin, checkMaxAttendees
```
Comment (OmTheTurtle) [inline src/components/groups/group.routes.ts]: Let's move this to a new line.

**Review by OmTheTurtle — CHANGES_REQUESTED** [review]
When dealing with default values, it's always safer to do it on database level. You can add a `.defaultTo(100)` to the migration, and with that you can remove the `|| 100` from the rest of the application.

File: views/group/show.pug [hunk 5]
```diff
@@ -93,8 +93,8 @@ block content
     table.table.is-fullwidth.is-hoverable
       thead
         .columns
-          .column.menu-label Résztvevők
-          if !joined && !group.doNotDisturb
+          .column.menu-label Résztvevők !{group.users.length} / !{group.maxAttendees || 100}
```
Comment (OmTheTurtle) [inline views/group/show.pug:96]: It's preferable to use the `#{variable}` syntax, as it escapes any potential text. Also, please remove the `|| 100` expressions here and in line 97 as they're not needed with the db-level default.

**Comment by OmTheTurtle** [comment]
You have a file conflict with the latest changes on master, please rebase your branch and I'll approve.

**Review by triszt4n — APPROVED** [review]
Cool and good.

**Review by OmTheTurtle — APPROVED** [review]

**Pull request merged**